File: 프로그래머스/lv2/12985. 예상 대진표/예상 대진표.js
function solution(n,a,b){
  let teamA = Math.round(a / 2)
  let teamB = Math.round(b / 2)
  let big = teamA > teamB ? teamA : teamB;
  let small = teamA < teamB ? teamA : teamB;
  let round = 1;
  if(teamA === teamB) {
      return 1;
  }
  while(true) {
      round++;
      if(big - small === 1 && big % 2 === 0) {
          break;
      }
      teamA = Math.round(teamA / 2);
      teamB = Math.round(teamB / 2);
      big = teamA > teamB ? teamA : teamB;
      small = teamA < teamB ? teamA : teamB;
  }
  
  return round;
}

File: 프로그래머스/lv2/12985. 예상 대진표/예상 대진표.java
class Solution
{
    public int solution(int n, int a, int b)
    {
       int teamA = a;
       int teamB = b;
       int round = 0;
        
       while(true) {
           round++;
           if(Math.abs(teamA - teamB) == 1 && Math.max(teamA,teamB) % 2 == 0) {
               break;
           }
            teamA = (int) Math.ceil((double) teamA / 2);
            teamB = (int) Math.ceil((double) teamB / 2);
       }
      return round;
    }
}